hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/table/functional/TestCleanActionExecutor.java

```diff
@@ -0,0 +1,156 @@
+import org.apache.hudi.exception.HoodieException;
+import org.apache.hudi.storage.HoodieStorage;
+import org.apache.hudi.storage.StorageConfiguration;
+import org.apache.hudi.table.HoodieTable;
+import org.apache.hudi.table.action.clean.CleanActionExecutor;
+import org.apache.hudi.table.action.clean.CleanPlanner;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.apache.hudi.common.testutils.HoodieTestUtils.getDefaultStorageConf;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests Clean action executor.
+ */
+public class TestCleanActionExecutor {
+
+  private static final StorageConfiguration<Configuration> CONF = getDefaultStorageConf();
+  private final HoodieEngineContext context = new HoodieLocalEngineContext(CONF);
+  private final HoodieTable<?, ?, ?, ?> mockHoodieTable = mock(HoodieTable.class);
+  private HoodieTableMetaClient metaClient;
+  private FileSystem fs;
+
+  private static String PARTITION1 = "partition1";
+
+  String earliestInstant = "20231204194919610";
+  String earliestInstantMinusThreeDays = "20231201194919610";
+
+  @BeforeEach
+  void setUp() {
+    metaClient = mock(HoodieTableMetaClient.class);
+    when(mockHoodieTable.getMetaClient()).thenReturn(metaClient);
+    HoodieTableConfig tableConfig = new HoodieTableConfig();
+    when(metaClient.getTableConfig()).thenReturn(tableConfig);
+    HoodieStorage storage = mock(HoodieStorage.class);
+    when(metaClient.getStorage()).thenReturn(storage);
+    when(mockHoodieTable.getStorage()).thenReturn(storage);
+    fs = mock(FileSystem.class);
+    when(storage.getFileSystem()).thenReturn(fs);
+    when(fs.getConf()).thenReturn(CONF.unwrap());
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  void testPartialCleanFailure(boolean simulateFailedDeletion) throws IOException {
+    HoodieWriteConfig config = getCleanByCommitsConfig();
+    String fileGroup = UUID.randomUUID() + "-0";
+    HoodieBaseFile baseFile = new HoodieBaseFile(String.format("/tmp/base/%s_1-0-1_%s.parquet", fileGroup, "001"));
+    FileSystem localFs = new Path(baseFile.getPath()).getFileSystem(CONF.unwrap());
+    Path filePath = new Path(baseFile.getPath());
+    localFs.create(filePath);
+    String exceptionMsg = "throwing run time exception";
+    if (simulateFailedDeletion) {
+      when(fs.delete(filePath, false)).thenThrow(new RuntimeException(exceptionMsg));
```

Comment:
Should `fs.delete` return false to cover the added logic?  If `fs.delete` throws an exception, does the cleaner already fail on master (does the tests still pass without the fix on master)?

Reply:
added more tests to cover all diff scenarios